synapse/rest/media/v1/_base.py

```diff
@@ -69,13 +70,30 @@ def parse_media_id(request):
         )
 
 
-def respond_404(request):
-    respond_with_json(
-        request,
-        404,
-        cs_error("Not found %r" % (request.postpath,), code=Codes.NOT_FOUND),
-        send_cors=True,
-    )
+def respond_404(request: Request):
+    respond_with_code(request, 404)
+
+
+def respond_with_code(request: Request, code: int):
```

Comment:
fwiw, the thumbnail API returning JSON for errors is expected under the Standard Error Response.

Reply:
That really doesn't makes semantic sense for an API, but if that's what the spec says I can revert this change.